src/com/teamcautionrobotics/robot2018/Tower.java

```diff
@@ -0,0 +1,164 @@
+package com.teamcautionrobotics.robot2018;
+
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.VictorSP;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.PIDController;
+import edu.wpi.first.wpilibj.PIDOutput;
+import edu.wpi.first.wpilibj.PIDSource;
+import edu.wpi.first.wpilibj.PIDSourceType;
+
+public class Tower implements PIDOutput, PIDSource {
+
+    enum TowerPosition {
+        GROUND, SWITCH, LOW_SCALE, HIGH_SCALE;
+
+        private static TowerPosition[] vals = values();
+
+        public TowerPosition next() {
+            if (this.ordinal() == vals.length - 1) {
+                return vals[this.ordinal()];
+            } else {
+                return vals[this.ordinal() + 1];
+            }
+        }
+
+        public TowerPosition previous() {
+            if (this.ordinal() == 0) {
+                return vals[this.ordinal()];
+            } else {
+                return vals[this.ordinal() - 1];
+            }
+        }
+    }
+
+    private TowerPosition currentTowerPosition;
+
+    private VictorSP towerMotor;
+    private Encoder towerEncoder;
+    private PIDController pidController;
+
+    private double desiredPosition;
+
+    public Tower(int motorPort, int encoderChannelA, int encoderChannelB, double Kp, double Ki,
+            double Kd) {
+        towerMotor = new VictorSP(motorPort);
+        towerEncoder = new Encoder(encoderChannelA, encoderChannelB);
+        towerEncoder.setDistancePerPulse((4 * Math.PI) / 1024);
+        pidController = new PIDController(Kp, Ki, Kd, 0, this, this);
+        pidController.setOutputRange(-1, 1);
+        pidController.setAbsoluteTolerance(3);
+    }
+
+    /**
+     * @param power positive is ascending, negative is descending, range of [-1, 1]
+     */
+    public void move(double power) {
+        towerMotor.set(power);
+        disablePID();
+    }
+
+    public void ascend() {
+        this.move(1.0);
+    }
+
+    public void descend() {
+        this.move(-1.0);
+    }
+
+    public void stop() {
+        this.move(0);
+    }
+
+    public void setPosition(TowerPosition desiredTowerPosition) {
+        switch (desiredTowerPosition) {
```

Comment:
If you add the height of the levels to TowerPosition, you can remove this switch statement.